administrator/components/com_modules/src/Model/ModuleModel.php

```diff
@@ -1163,4 +1209,108 @@ protected function cleanCache($group = null, $clientId = 0)
 	{
 		parent::cleanCache('com_modules');
 	}
+
+	/**
+	 * Method to change the published state of one or more records.
+	 *
+	 * @param   array    &$pks   A list of the primary keys to change.
+	 * @param   integer  $value  The value of the published state.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function publish(&$pks, $value = 1)
+	{
+		$this->workflowBeforeStageChange();
+
+		return parent::publish($pks, $value);
+	}
+
+	/**
+	 * Overrides the function in WorkflowBehaviorTrait to get the default
+	 * stage ID while creating a new module.
+	 *
+	 * @param   Form   $form  A Form object.
+	 * @param   mixed  $data  The data expected for the form.
+	 *
+	 * @return  boolean|integer  An integer, holding the stage ID or false
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getStageForNewItem(Form $form, $data)
+	{
+		$db = $this->getDbo();
+
+		$query = $db->getQuery(true)
+			->select($db->quoteName('id'))
+			->from($db->quoteName('#__workflow_stages'))
+			->where($db->quoteName('default') . '= 1');
+
+		if (isset($data['workflow_id']))
+		{
+			// If workflow_id is set by the query then we find the stage corresponding to that workflow_id
+			$query->where($db->quoteName('workflow_id') . ' = :workflowid')
+				->bind(':workflowid', $data['workflow_id'], ParameterType::INTEGER);
+		}
+		else
+		{
+			// Query to find the default workflow id of the extension type com_modules.module
+			$defaultWorkflowQuery = $db->getQuery(true)
+				->select($db->quoteName('id'))
+				->from($db->quoteName('#__workflows'))
+				->where($db->quoteName('default') . ' = 1')
+				->where($db->quoteName('extension') . ' = "com_modules.module"');
```

Comment:
```suggestion
				->where($db->quoteName('extension') . ' = ' . $db->quote('com_modules.module');
```
Same issue as the other one before.

Reply:
Thank you once again! :)

Reply:
And there are more